client/src/controllers/PreviewController.ts

```diff
@@ -96,6 +97,13 @@ export class PreviewController extends Controller<HTMLElement> {
   /** The device size width to use when the preview is not available. */
   static fallbackWidth = PREVIEW_UNAVAILABLE_WIDTH.toString();
 
+  /**
+   * The time tolerance between the iframe's `load` event and the scroll
+   * restoration completion, which may not be instantaneous for cross-domain
+   * preview iframes.
+   */
+  static scrollRestoreTimeout = 10_000; // 10 seconds
```

Comment:
What do you think this value should be? It's only for the time it takes to complete the `postMessage` ping-pong, not the time for the iframe to fire the load event. However, in a headless setup, the userbar is usually loaded _after_ the page has been loaded, so we can't expect a value like 1s or lower.

The problem with a high number is that this will block the new iframe from being displayed, because we only want to display it after the scroll has finished (to avoid the jump being visible).